### PR TITLE
[Bloganuary] Dashboard Nudge card

### PR DESCRIPTION
**Pull request by thomashorta**
Part of #19663 

This only contains the UI part of the Bloganuary Nudge Card and basic logic to show it on the dashboard, the interactions are still TODOs and will be done in different PRs.

Light Mode | Dark Mode
--- | ---
bloganuary-nudge-light-mode | bloganuary-nudge-dark-mode

-----

## To Test:

1. Install and log into Jetpack (with a site that can see Blogging Prompts)
2. Go to Debug settings
3. Confirm the `bloganuary_dashboard_nudge` FF is ON (should be by default)
4. Go to My Site Dashboard
5. **Verify** the "Bloganuary is coming" card is shown (the buttons/interactions still don't work)
6. Go to Debug settings
7. Disable the `bloganuary_dashboard_nudge` FF
8. Go to My Site Dashboard
9. **Verify** the Bloganuary card is not shown

-----

## Regression Notes

1. Potential unintended areas of impact

    - Other cards UI/behavior affected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

10. What automated tests I added (or what prevented me from doing so)

    - Unit tests for Bloganuary card builder
    - Unit tests for Bloganuary ViewModel slice
    - Updated unit tests for existing modified classes

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:

- [x] Portrait and landscape orientations.
- [x] Light and dark modes.
- [x] Fonts: Larger, smaller and bold text.
- [x] High contrast.
- [x] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt [hunk 1]
```diff
@@ -751,6 +751,11 @@ class MySiteFragment : Fragment(R.layout.my_site_fragment),
         is SiteNavigationAction.OpenDashboardPersonalization -> activityNavigator.openDashboardPersonalization(
             requireActivity()
         )
+
+        is SiteNavigationAction.OpenBloganuaryNudgeOverlay -> {
+            // TODO thomashortadev: Implement Bloganuary nudge overlay
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt:756]: Just to confirm: is this `TODO`/`AppLog` part of another PR or should we remove it?

File: WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewModelSliceTest.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        val params = viewModel.getBuilderParams()
+        params.onLearnMoreClick.invoke()
+
+        advanceUntilIdle()
+        assertThat(viewModel.onNavigation.value?.peekContent()).isEqualTo(
+            SiteNavigationAction.OpenBloganuaryNudgeOverlay(isPromptsEnabled)
+        )
+    }
+
+    // TODO thomashortadev: test onMoreMenuClick and onHideMenuItemClick when they are implemented
```
Comment (RenanLukas) [inline WordPress/src/test/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewModelSliceTest.kt:79]: Also just to confirm: this is going to be part of a future PR, right?

File: WordPress/src/test/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsSettingsHelperTest.kt [hunk 3]
```diff
@@ -193,9 +192,8 @@ class BloggingPromptsSettingsHelperTest : BaseUnitTest() {
         }
 
 
-    @Suppress("MaxLineLength")
     @Test
-    fun `given prompts setting on and not skipped for today, when isPromptsFeatureActive, then returns false`() =
+    fun `given prompts setting on and not skipped for today, when shouldShowPromptsFeature, then returns true`() =
```
Comment (RenanLukas) [inline WordPress/src/test/java/org/wordpress/android/ui/bloggingprompts/BloggingPromptsSettingsHelperTest.kt:196]: Good catch 👍

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewModelSlice.kt [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    val onNavigation = _onNavigation as LiveData<Event<SiteNavigationAction>>
+
+    private lateinit var scope: CoroutineScope
+
+    fun initialize(scope: CoroutineScope) {
+        this.scope = scope
+    }
+
+    fun getBuilderParams(): BloganuaryNudgeCardBuilderParams {
+        // TODO thomashortadev: check if current device date is in December 2023
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewModelSlice.kt:28]: Same question about the TODOs in this class (if they should be removed or if they will be handled by another PR)

Also I wonder if it's possible for us to use the server date instead.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewHolder.kt [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import org.wordpress.android.ui.compose.theme.AppTheme
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.BloganuaryNudgeCardModel
+import org.wordpress.android.ui.mysite.MySiteCardAndItemViewHolder
+import org.wordpress.android.util.extensions.viewBinding
+
+class BloganuaryNudgeCardViewHolder(parent: ViewGroup) :
+    MySiteCardAndItemViewHolder<BloganuaryNudgeCardBinding>(parent.viewBinding(BloganuaryNudgeCardBinding::inflate)) {
+    fun bind(cardModel: BloganuaryNudgeCardModel) = with(binding.bloganuaryNudgeCard) {
+        // Dispose of the Composition when the view's LifecycleOwner is destroyed
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnDetachedFromWindowOrReleasedFromPool)
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewHolder.kt:17]: TIL nice

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCard.kt [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    ) {
+        Column {
+            CardToolbar(model)
+
+            Spacer(Modifier.height(8.dp))
+
+            Text(
+                text = uiStringText(model.title),
+                style = DashboardCardTypography.subTitle,
+                modifier = Modifier.padding(horizontal = 16.dp),
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCard.kt]: 💡 Would it make sense to use `Margin.ExtraLarge` instead?

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCard.kt [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+            )
+
+            Spacer(Modifier.height(4.dp))
+
+            Text(
+                text = uiStringText(model.text),
+                style = DashboardCardTypography.detailText,
+                modifier = Modifier.padding(horizontal = 16.dp),
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCard.kt]: 💡 Same question about using `Margin`

**Review by RenanLukas — APPROVED** [review]
Thanks for the PR. I've left some optional suggestions and I'm approving it.

**Review by osullivanchris — APPROVED** [review]
still having trouble logging into builds but from the screenshots this looks good 👍

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteFragment.kt:756]: It is part of another PR, I'm keeping it since the work is being done in the feature branch.

File: WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewModelSlice.kt (hunk 4 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/bloganuary/BloganuaryNudgeCardViewModelSlice.kt:28]: The blogging prompt logic to get the current-day prompt uses the local device date, so it makes sense to use the device date here as well as that's when the prompt will change anyway.

The TODOs I left in code were intentional since the feature is in-development and there are a few empty methods.

**Review by RenanLukas — APPROVED** [review]
Thanks for implementing the suggestions. LGTM :shipit:
